c7n/resources/s3.py

```diff
@@ -1372,6 +1372,158 @@ def process_bucket(self, bucket):
                 return {'Name': bucket['Name'], 'State': 'ReplicationConfigUpdated'}
 
 
+@filters.register('check-public-block')
+class FilterPublicBlock(Filter):
+    """Filter for s3 bucket public blocks
+
+    `scope`: Optional: Defaults to Any
+    `enabled`: Optional: Defaults to False
+
+    The defaults essentially check to see if any public blocks are missing
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: CheckForPublicAclBlock-Off
+                resource: s3
+                region: us-east-1
+                filters:
+                  - type: check-public-block
+                    scope: Any
+                    enabled: False
+
+    """
+
+    schema = type_schema(
+        'check-public-block',
+        scope={
+            'type': 'string',
+            'enum': ['BlockPublicAcls', 'IgnorePublicAcls',
+                'BlockPublicPolicy', 'RestrictPublicBuckets', 'All', 'Any']},
+        enabled={'type': 'boolean'})
+    permissions = ("s3:GetBucketPublicAccessBlock",)
+
+    def process(self, buckets, event=None):
+        results = list()
+        with self.executor_factory(max_workers=2) as w:
+            futures = {w.submit(self.process_bucket, bucket): bucket for bucket in buckets}
+            for future in as_completed(futures):
+                if future.exception():
+                    raise future.exception()
+                if future.result():
+                    results.append(future.result())
+        return results
+
+    def process_bucket(self, bucket):
+        s3 = bucket_client(local_session(self.manager.session_factory), bucket)
+        enabled = self.data.get('enabled', False)
+        scope = self.data.get('scope', 'Any')
+        try:
+            config = s3.get_public_access_block(
+                Bucket=bucket['Name'])['PublicAccessBlockConfiguration']
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'NoSuchPublicAccessBlockConfiguration':
+                config = None
+            else:
+                raise Exception(e)
```

Comment:
this one should just raise instead of turning this into a generic exception, and right above instead of doing config=None just return

Reply:
So here, I'll add a note about it but if all public blocks are set to false on a bucket, the get_public_access_block call returns a 'NoSuchPublicAccessBlockConfiguration' error. I am setting to None so that later on I can do the if else block in matches_filter on line 1434

Reply:
Notes added in a4bf33182c9d128df2a0115168466b166abf42d6

Reply:
yeah.. the config stuff looks fine, thanks

Reply:
Thanks! It was kind of weird to me too, seems like it would have been just as easy for that call to return all falses. I guess setting to None/null probably saves some size on the storage though since AWS has been defaulting to all public blocks on new account creation.